module/x/gravity/keeper/keeper.go

```diff
@@ -369,15 +373,31 @@ func (k Keeper) GetOrchestratorValidator(ctx sdk.Context, orch sdk.AccAddress) s
 /////////////////////////////
 
 // SetEthAddress sets the ethereum address for a given validator
-func (k Keeper) SetEthAddress(ctx sdk.Context, validator sdk.ValAddress, ethAddr string) {
+func (k Keeper) SetEthAddressForValidator(ctx sdk.Context, validator sdk.ValAddress, ethAddr string) {
 	store := ctx.KVStore(k.storeKey)
-	store.Set(types.GetEthAddressKey(validator), []byte(ethAddr))
+	store.Set(types.GetEthAddressByValidatorKey(validator), []byte(ethAddr))
+	store.Set(types.GetValidatorByEthAddressKey(ethAddr), []byte(validator))
 }
 
-// GetEthAddress returns the eth address for a given gravity validator
-func (k Keeper) GetEthAddress(ctx sdk.Context, validator sdk.ValAddress) string {
+// GetEthAddressByValidator returns the eth address for a given gravity validator
+func (k Keeper) GetEthAddressByValidator(ctx sdk.Context, validator sdk.ValAddress) string {
 	store := ctx.KVStore(k.storeKey)
-	return string(store.Get(types.GetEthAddressKey(validator)))
+	return string(store.Get(types.GetEthAddressByValidatorKey(validator)))
+}
+
+// GetValidatorByEthAddress returns the validator for a given eth address
+func (k Keeper) GetValidatorByEthAddress(ctx sdk.Context, ethAddr string) (validator stakingtypes.Validator, found bool) {
+	store := ctx.KVStore(k.storeKey)
+	valAddr := store.Get(types.GetValidatorByEthAddressKey(ethAddr))
+	if valAddr == nil {
+		return stakingtypes.Validator{}, false
+	}
+	validator, found = k.StakingKeeper.GetValidator(ctx, valAddr)
+	if !found {
+		return stakingtypes.Validator{}, false
+	}
+
+	return validator, true
```

Comment:
I'd highly recommend checking the Uber golang style guide for reference: https://github.com/uber-go/guide

```suggestion
	return k.StakingKeeper.GetValidator(ctx, valAddr)
```

Reply:
Are you referring to the use of the "found" boolean? I pinged you about this in Telegram a few days ago. It was tough to decide whether or not to use the "found" boolean, since it seems that the Cosmos-SDK does not use a consistent pattern. I settled on using "found" since at least it is more explicit.